assets/js/schedule.js

```diff
@@ -55,12 +71,26 @@ function createHourBlock(hour, selectedDay, selectedMonth, selectedYear, storedA
 }
 
 function getStoredAppointments(day, month, year) {
-    const storedData = localStorage.getItem('appointments');
-    const listAppointments = storedData ? JSON.parse(storedData) : [];
-
+    const listAppointments = getStorageAppointments();
     return listAppointments
         .filter((appointment) => {
             return appointment.day === day && appointment.month === month && appointment.year === year;
         });
+}
+
+function getStorageAppointments() {
+    const storedData = localStorage.getItem('appointments');
+    return storedData ? JSON.parse(storedData) : [];    
+}
+
+function cancelAppointment(day, month, year, hour) {
+    let listAppointments = getStorageAppointments();
+    const updatedAppointments = listAppointments.filter(appointment =>
+        !(appointment.day === day && appointment.month === month && appointment.year === year && appointment.hour == hour)
+    );
+
+    localStorage.setItem('appointments', JSON.stringify(updatedAppointments));
 
+    alert(`Appointment at ${hour}:00 successfully canceled!`);
```

Comment:
Using `alert` to inform users is not a good practice. This is being used throughout the application, so I'll create an issue to handle that.